client/extensions/woocommerce/app/promotions/promotion-models.js

```diff
@@ -108,46 +106,48 @@ const appliesToProductSale = {
 	labelText: translate( 'Applies to product' ),
 	cssClass: 'promotions__promotion-form-card-applies-to',
 	fields: {
-		appliesTo: {
-			component: (
-				<PromotionAppliesToField
-					selectionTypes={ [ { type: 'productIds' } ] }
-					singular={ true }
-				/>
-			),
-			// TODO: Remove this text after variable products are supported.
-			explanationText: translate(
-				'Note: Variable products cannot be selected.'
-			),
-		},
+		appliesTo: (
+			<PromotionAppliesToField
+				selectionTypes={ [ { type: 'productIds' } ] }
+				singular={ true }
+				// TODO: Remove this text after variable products are supported.
+				explanationText={ translate( 'Note: Variable products cannot be selected.' ) }
+			/>
+		),
 	},
 };
 
 /**
- * Field reused for fixed currency-based coupons.
+ * End date condition field
+ *
+ * @param { Object } props Component properties
+ * @param { Object } props.promotion Promotion API object
+ * @param { string } props.value Current value for end date.
+ * @return { Object } React component instance.
  */
-const fixedDiscountField = {
-	labelText: translate( 'Discount', { context: 'noun' } ),
-	component: CurrencyField,
-	isRequired: true,
-};
+const EndDateField = ( props ) => {
+	const { promotion, value } = props;
+	const startDate = promotion.startDate ? new Date( promotion.startDate ) : new Date();
+	const endDate = value ? new Date( value ) : new Date();
+	let validationError = null;
 
-/**
- * General purpose start date condition.
- */
-const startDate = {
-	component: DateField,
-	labelText: translate( 'Start Date' ),
-	isEnableable: true,
-};
+	// Clear the times, we only care about days.
+	startDate.setHours( 0, 0, 0, 0 );
+	endDate.setHours( 0, 0, 0, 0 );
 
-/**
- * General purpose end date condition.
- */
-const endDate = {
-	component: DateField,
-	labelText: translate( 'End Date' ),
-	isEnableable: true,
+	if ( endDate < startDate ) {
```

Comment:
Do we allow endDate === startDate?

Reply:
Yeah, I don't know of any other way to enable a 1-day sale.